### PR TITLE
Add prettier for codestyle and re-format everything

### PR DESCRIPTION
**Pull request by ErikMichelson**
### Component/Part
meta, reformats everything

### Description
This PR adds [prettier](https://prettier.io) as a codestyle enforcer and formatter like the backend repo does. 
This should result in the same clean code over all files. To ensure that prettier is used, husky + lint-staged are included in this PR as well, which run prettier as a pre-commit hook. This is recommended by prettier themselves. Another advantage is that prettier works as a dev-dependency in every node-environment while our editorconfig file only works in JetBrains products.

### Steps

- [x] Added implementation
- [x] I read the [contribution documentation](https://github.com/hedgedoc/react-client/blob/main/CONTRIBUTING.md) and signed-off my commits to accept the DCO.

### Related Issue(s)
none

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -144,9 +148,21 @@
...
+  "lint-staged": {
+    "src/**/*.{ts,tsx,js}": "prettier -w --ignore-unknown"
+  },
+  "prettier": {
+    "parser": "typescript",
+    "singleQuote": true,
+    "jsxSingleQuote": true,
+    "semi": false,
+    "tabWidth": 2,
+    "trailingComma": "none"
```
Comment (DerMolly) [inline package.json:164]: This seems to remove many commas we wrote, maybe this option should not be `none`?

Comment (mrdrogdrog) [reply package.json:164]: If you check our `.editorconfig` then you'll see that we set `ij_typescript_enforce_trailing_comma = remove`. So I guess we always wanted to remove the commas.

File: package.json [hunk 2]
```diff
@@ -168,9 +184,13 @@
     "cypress": "7.4.0",
     "cypress-commands": "1.1.0",
     "cypress-file-upload": "5.0.7",
+    "eslint-config-prettier": "^8.3.0",
     "eslint-plugin-chai-friendly": "0.7.1",
     "eslint-plugin-cypress": "2.11.3",
     "http-server": "0.12.3",
+    "husky": "^6.0.0",
+    "lint-staged": "^11.0.0",
+    "prettier": "^2.3.0",
```
Comment (mrdrogdrog) [inline package.json:193]: pin

File: src/components/application-loader/initializers/i18n.ts [hunk 3]
```diff
@@ -19,7 +19,7 @@ export const setUpI18n = async (frontendAssetsUrl: string): Promise<void> => {
       fallbackLng: 'en',
       debug: process.env.NODE_ENV !== 'production',
       backend: {
-        loadPath: `${ frontendAssetsUrl }locales/{{lng}}.json`
+        loadPath: `${frontendAssetsUrl}locales/{{lng}}.json`
```
Comment (mrdrogdrog) [inline src/components/application-loader/initializers/i18n.ts:22]: This seems to be wrong. IIRC we WANT these spaces

**Comment by DerMolly** [comment]
I'd suggest adding the following to the prettier config:

```
bracketSpacing: true,
jsxBracketSameLine: true,
arrowParens: 'always',
```

File: package.json (hunk 1 above)
Comment (DerMolly) [reply package.json:164]: Then I withdraw my initial comment…

File: src/components/application-loader/initializers/i18n.ts (hunk 3 above)
Comment (ErikMichelson) [reply src/components/application-loader/initializers/i18n.ts:22]: That's right. I still have to find an option for that though.

Comment (DerMolly) [reply src/components/application-loader/initializers/i18n.ts:22]: Apparently there is no option for this and https://github.com/prettier/prettier/issues/95 suggests prettier won't change this.

**Review by DerMolly — APPROVED** [review]

**Review by mrdrogdrog — APPROVED** [review]

**Pull request merged**